lib/rake/testtask.rb

```diff
@@ -116,8 +116,9 @@ def define
             if !ok && status.respond_to?(:signaled?) && status.signaled?
               raise SignalException.new(status.termsig)
             elsif !ok
-              fail "Command failed with status (#{status.exitstatus}): " +
-                "[ruby #{args}]"
+              status  = "Command failed with status (#{status.exitstatus})"
+              details = ": [ruby #{args}]"
+              fail (ARGV.include?('--trace') || @verbose) ? (status + details) : status
```

Comment:
`ARGV` may no longer include `'--trace'` by the time we get here (Rakefile authors may change it).  Switching to `Rake.application.options.trace` will keep this value.

Also, can you do one thing per line so this is easier to maintain?  Like this:

```ruby
message =
  if Rake.application.options.trace or @verbose then
    status + details
  else
    status
  end

fail message
```